openvpn/config_validator.go

```diff
@@ -0,0 +1,122 @@
+package openvpn
+
+import (
+	"bufio"
+	"bytes"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"github.com/mysterium/node/session"
+	"net"
+	"strings"
+)
+
+// ValidateConfig is function which takes VPNConfig as argument, checks it and returns error if validation fails
+type ValidateConfig func(config session.VPNConfig) error
+
+// ConfigValidator represents structure which contains list of validating functions
+type ConfigValidator struct {
+	validators []ValidateConfig
+}
+
+// NewDefaultValidator returns ConfigValidator with predefined list of validating functions
+func NewDefaultValidator() *ConfigValidator {
+	return &ConfigValidator{
+		validators: []ValidateConfig{
+			validProtocol,
+			validPort,
+			validIPFormat,
+			validTLSPresharedKey,
+			validCACertificate,
+		},
+	}
+}
+
+// IsValid function checks if provided config is valid against given config validator and returns first encountered error
+func (v *ConfigValidator) IsValid(config session.VPNConfig) error {
+	for _, validator := range v.validators {
+		if err := validator(config); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func validProtocol(config session.VPNConfig) error {
+	switch config.RemoteProtocol {
+	case
+		"udp",
+		"tcp":
+		return nil
+	}
+	return errors.New("invalid protocol: " + config.RemoteProtocol)
+}
+
+func validPort(config session.VPNConfig) error {
+	if config.RemotePort > 65535 || config.RemotePort < 1 {
```

Comment:
should be 'config.RemotePort < 1024', first 1023 ports are reserved for superUser processes and cannot be used with userSpace processes.